### PR TITLE
(GH-1426) - Update value to accept array

### PR DESCRIPTION
**Pull request by jordanbreen28**
Fixes https://github.com/puppetlabs/puppetlabs-postgresql/issues/1426

### CODE REVIEW

**Review by GSPatton — APPROVED** [review]

File: manifests/server/config_entry.pp [hunk 1]
```diff
@@ -5,9 +5,9 @@
 # @param path Path for postgresql.conf
 #
 define postgresql::server::config_entry (
-  Enum['present', 'absent']             $ensure = 'present',
-  Optional[Variant[String[1], Numeric]] $value  = undef,
-  Variant[Boolean, String[1]]           $path   = false
+  Enum['present', 'absent']                               $ensure = 'present',
+  Optional[Variant[String[1], Numeric, Array[String[1]]]] $value  = undef,
```
Comment (ekohl) [inline manifests/server/config_entry.pp:9]: Shouldn't this also an array of `Numeric`?

**Review by ekohl — COMMENTED** [review]
Not objecting to having merged this, but I only just got the notification (something's up with GH). It would have been nice to add a (regression) test.

File: manifests/server/config_entry.pp (hunk 1 above)
Comment (jordanbreen28) [reply manifests/server/config_entry.pp:9]: @ekohl I think that `Array[String[1]]` is suitable in the case of passing config values for listen_addresses (i understand value , curious on your thoughts of updating this to accept `Array[Numeric]` values as well.

I think continuing to update this is a bit out of scope, as initially this was implemented as a quick bug fix to ensure that existing manifests with values for `listen_addresses` did not break after the syntax update carried out on this module. but I'm open to your suggestions. 💯

Comment (ekohl) [reply manifests/server/config_entry.pp:9]: It's probably fine, since I don't know how many array values there are where it is numeric. You can indeed use the following in YAML as a workaround:
```yaml
---
value:
  - '1'
  - '2'
```

But very often you see some consistency, like `Variant[X, Y, Array[Variant[X, Y]]]` so that's something that always goes through my head. Still, it's not always true. For example, the current type can be written as `Variant[Undef, String[1], Numeric, Array[String[1]]]` (which is actually shorter than `Optional[]`) and you don't want to accept `Array[Undef]`.

That's at least how I look at these data types and assess if it fits what's needed.

The config entry is supposed to reflect a setting, so https://www.postgresql.org/docs/current/config-setting.html is a good resource. That states:

> All parameter names are case-insensitive. Every parameter takes a value of one of five types: boolean, string, integer, floating point, or enumerated (enum).

So String and Numeric certainly are good. Enum is covered by String. Then for booleans there's:

> Boolean: Values can be written as on, off, true, false, yes, no, 1, 0 (all case-insensitive) or any unambiguous prefix of one of these.

So accepting `Boolean` is also good.

There's no real array type, but there are types which are lists. https://www.postgresql.org/docs/current/runtime-config.html is quite long. Most are comma joined with `ssl_ciphers` as an exception. We can ignore that one and require the user to specify it as a single string.

In conclusion: I think the current type will work.

Comment (jordanbreen28) [reply manifests/server/config_entry.pp:9]: @ekohl as always, thanks for your great work and investigation into keeping our modules in top condition, it is appreciated!!

Completely understand your reasoning now thanks to all that info, I'll add a ticket to our backlog to further investigate this and implement it if required/we think it will be nice (which I must admit I do now..).

So thanks again 💯
